demux/utils/novaseq_samplesheet.py

```diff
@@ -0,0 +1,181 @@
+            dummy_samples = [row for row in dummy_samples_csv]
+            added_dummy_samples = []
+            lanes = {sample["lane"] for sample in self.raw_samplesheet}
+
+            for lane in lanes:
+                sample_indexes = [
+                    sample["index"]
+                    for sample in self.raw_samplesheet
+                    if sample["lane"] == lane
+                ]
+                for name, dummy_index in dummy_samples:
+                    if not (
+                        any(
+                            sample_index.startswith(dummy_index)
+                            for sample_index in sample_indexes
+                        )
+                    ):
+                        add_dummy_sample = {
+                            "control": "N",
+                            "description": "",
+                            "fcid": self.flowcell,
+                            "index": dummy_index,
+                            "index2": "",
+                            "lane": lane,
+                            "operator": "script",
+                            "project": "indexcheck",
+                            "recipe": "R1",
+                            "sample_id": name.replace(" ", "-")
+                            .replace("(", "-")
+                            .replace(")", "-"),
+                            "sample_name": "indexcheck",
+                            "sample_ref": "hg19",
+                        }
+
+                        added_dummy_samples.append(add_dummy_sample)
+
+            self.raw_samplesheet.extend(added_dummy_samples)
+
+            return self
+
+    def remove_unwanted_indexes(self):
+        """ Filter out indexes of unwanted length and single indexes """
+
+        if self.indexlength:
+            if self.pad and int(self.indexlength) in (16, 20):
+                self.raw_samplesheet = [
+                    line
+                    for line in self.raw_samplesheet
+                    if len(line["index"].replace("-", ""))
+                    in (16, int(self.indexlength))
+                ]
+            else:
+                self.raw_samplesheet = [
+                    line
+                    for line in self.raw_samplesheet
+                    if len(line["index"].replace("-", "")) == int(self.indexlength)
+                ]
+        else:
+            self.raw_samplesheet = [
+                line for line in self.raw_samplesheet if "-" in line["index"]
+            ]
+
+        return self
+
+    def adapt_indexes(self):
+        """Adapts the indexes: pads all indexes so that all indexes have a length equal to the
+        number  of index reads, and takes the reverse complement of index 2 in case of the new
+        novaseq software control version (1.7) in combination with the new reagent kit
+        (version 1.5)"""
+
+        is_reverse_complement = (
+            self.runparameters.control_software_version == "1.7.0"
```

Comment:
Looks like you should use named constants for "1.7.0" and "1.5" - to explain what they represent